### PR TITLE
[artifactory,distribution,insight,pipelines,xray] Add annotations on statefulsets and deployments

### PR DESCRIPTION
**Pull request by bmanuel**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
Adds support for annotations on statefulsets and deployments in addition to the pod specs that were already present. This is helpful when interacting with other tooling on the deployment clusters that function via annotations.

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by bmanuel** [comment]
recheckcla

**Comment by bmanuel** [comment]
recheck

**Comment by oumkale** [comment]
Hi @bmanuel, Thank you for your PR.

We will discuss internally and try to take it in upcoming released

**Comment by bmanuel** [comment]
Hey @oumkale, are there any updates on whether this will be pulled into future releases? Also, is it worth me keeping up with conflicts since the changes will be pulled into releases as the charts are updated?

**Comment by oumkale** [comment]
Hi @bmanuel,

We will take this in the next on-prem release, ETA ~3 weeks

**Comment by chukka** [comment]
@bmanuel  Can you please add these changes to `artifactory-ha` chart as well

File: stable/artifactory/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # JFrog Artifactory Chart Changelog
 All changes to this chart will be documented in this file.
 
+## [10.41.8] - 
+* Added support for annotations on the artifactory statefulset
```
Comment (chukka) [inline stable/artifactory/CHANGELOG.md:5]: @bmanuel  can you please add this  in just add one line
`Added support for annotations for artifactory statefulset and nginx deployment [GH-1665](https://github.com/jfrog/charts/pull/1665)`

**Comment by bmanuel** [comment]
@chukka artifactory-ha has been added.

**Comment by chukka** [comment]
@bmanuel This can you please squash all your commits to single one

**Comment by bmanuel** [comment]
@chukka since this is for multiple charts will it be easier to keep the commits separate? They are currently one chart per commit.

**Comment by chukka** [comment]
@bmanuel  Hmm, I think you are right, Can you please raise as separate PRs , we would like to acknowledge  your commits  from next releases and add it to your github opensource contributions list 

Sorry for pushing  this , Hope you understand :)

cc @pathob

**Comment by bmanuel** [comment]
@chukka, I will get this split up later today.

**Comment by bmanuel** [comment]
@chukka I split up this PR and linked the new ones to this PR. I am going to close this now in favor of those PRs.

**Comment by bmanuel** [comment]
@chukka is it possible to get the split up PRs reviewed and queued for release?

**Comment by chukka** [comment]
@bmanuel These PRs are already taken internally and will be part of upcoming releases of each product

**Pull request closed without merging**